rfcs/034-location_location_location/README.md

```diff
@@ -0,0 +1,403 @@
+*   The URL points to the IIIF manifest on `wellcomelibrary.org`.
+*   The license, access status and access conditions are drawn from the METS XML:
+
+    ```xml
+    <mods:accessCondition type="dz">CC-BY-NC</mods:accessCondition>
+    <mods:accessCondition type="status">Open</mods:accessCondition>
+    <mods:accessCondition type="usage">Some terms</mods:accessCondition>
+    ```
+
+Bibs and items from Sierra:
+
+*   We create a physical location whose location type comes from the `"location"` field on the Sierra API response for the item.
+*   The access conditions are based on the varfields with MARC tag 506:
+
+    -   If the first indicator is 0, the access status is "Open"
+    -   We look at subfield $a for a free-text description of the access conditions
+    -   We look at subfield $f for the standardised open/closed/restricted terminology
+
+    Note: if these three values are inconsistent, the code will not set an AccessStatus from our controlled vocabulary.
+    This is protection against future inconsistency; there are no known examples.
+
+
+
+## Proposed changes
+
+*   The AccessStatus model in the Catalogue API (which draws from a fixed vocabulary) should reflect the access statuses in section 12 of the [Wellcome Collection Access Policy](https://wellcomecollection.org/pages/Wvmu3yAAAIUQ4C7F#access-policy).
+
+    Currently it has: `Open`, `OpenWithAdvisory`, `Restricted`, `Unavailable`, `Closed`, `LicensedResources` and `PermissionRequired`.
+
+    We should add:
+
+    -   `ByAppointment`, which the current implementation of the Catalogue API treats as synonymous with `Restricted`
+    -   `TemporarilyUnavailable`, so we can distinguish between, say, material that's being digitised and will be available later, and material that has been de-accessioned and will never become available.
+
+*   Currently the `"license"` field only appears on digital locations.
+
+    Although we don't have any examples of it on physical locations yet, a license is sufficiently core that we'll make it part of the PhysicalLocation model.
+
+*   We will add an optional `"link_text"` field to digital locations.
+
+*   Currently the `LocationType` model uses the full range of location codes supported by Sierra.
+
+    We will replace this with a controlled set of types (similar to AccessStatus), and the Sierra transformer will make a best-effort guess to map the Sierra location code/label onto one of these types.
+    If it is unable to map the Sierra location code to one of these types, it will skip adding a location to the item.
+
+    This will include a warning log, so the developers can know if/when new locationTypes have been added and are being skipped by the transformer.
+
+    This is an initial suggestion for the list of types:
+
+    *   ClosedStores
+    *   OpenShelves
+    *   IIIFPresentationAPI
+    *   IIIFImageAPI
+    *   OnExhibition
+    *   OnlineResource
+    *   Conservation
```

Comment:
I'm not sure about this one. It's not a permanent state or one we have end for, like the others. Does it also apply to things that are at digitisation? Is this one is more of an availability status than a location type? It might make more sense to treat this the same as if it's in closed stores, because to a library user it effectively is.

Reply:
I've collapsed this into "ClosedStores with access status TemporarilyUnavailable", and rolled in digitisation as well.